commit 1bea9ab46d9f9871b71b078192699fc94287fcc4
Author: Thorsten Blum <thorsten.blum@linux.dev>
Date:   Mon Feb 24 22:54:50 2025 +0100

    MIPS: Loongson2ef: Replace deprecated strncpy() with strscpy()
    
    strncpy() is deprecated for NUL-terminated destination buffers. Use
    strscpy() instead and remove the manual NUL-termination.
    
    Compile-tested only.
    
    Link: https://github.com/KSPP/linux/issues/90
    Cc: linux-hardening@vger.kernel.org
    Signed-off-by: Thorsten Blum <thorsten.blum@linux.dev>
    Signed-off-by: Thomas Bogendoerfer <tsbogend@alpha.franken.de>

diff --git a/arch/mips/loongson2ef/common/machtype.c b/arch/mips/loongson2ef/common/machtype.c
index 82f6de49f20f..e635e66d2e6c 100644
--- a/arch/mips/loongson2ef/common/machtype.c
+++ b/arch/mips/loongson2ef/common/machtype.c
@@ -48,8 +48,7 @@ void __init prom_init_machtype(void)
 		return;
 	}
 	p += strlen("machtype=");
-	strncpy(str, p, MACHTYPE_LEN);
-	str[MACHTYPE_LEN] = '\0';
+	strscpy(str, p);
 	p = strstr(str, " ");
 	if (p)
 		*p = '\0';